packages/telemed-intake/app/src/pages/PatientPortal.tsx

```diff
@@ -3,19 +3,30 @@ import { Link } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
 import { IntakeFlowPageRoute } from '../App';
 import { otherColors } from '../IntakeThemeProvider';
-import { useGetAppointments } from '../features/appointments';
 import { CustomContainer, useIntakeCommonStore } from '../features/common';
 import HomepageOption from '../features/homepage/HomepageOption';
 import { useZapEHRAPIClient } from '../utils';
 import { requestVisit, pastVisits, contactSupport } from '@theme/icons';
 import { useGetPatients, usePatientsStore } from 'src/features/patients';
+import { useEffect, useState } from 'react';
+import { GetTelemedAppointmentsResponse } from 'ottehr-utils';
 
 const PatientPortal = (): JSX.Element => {
   localStorage.removeItem('welcomePath');
   const apiClient = useZapEHRAPIClient();
   const { t } = useTranslation();
 
-  const { data: appointmentsData, isFetching } = useGetAppointments(apiClient, Boolean(apiClient));
+  const [appointmentsData, setAppointmentsData] = useState<GetTelemedAppointmentsResponse | null>(null);
+  const [isFetching, setIsFetching] = useState(true);
+
+  useEffect(() => {
+    if (apiClient) {
+      void apiClient.getAppointments().then((data) => {
+        setAppointmentsData(data);
+        setIsFetching(false);
+      });
+    }
+  }, [apiClient]);
```

Comment:
Is there a specific reason for this change?

I recommend keeping the `useGetAppointments` hook instead of using direct `useEffect` fetching. Custom hooks provide better error handling, state management, and also good for handling cleanup and racing issues in a better way.

Reply:
useGetAppointments does not refresh when navigating back to the page, leading to bugs

Reply:
did you test this? I tried setting refetchOnMount: true and I encounter the same errors as before

Reply:
refactored to use useQuery like we had before